File: commons/src/main/java/com/ultreon/commons/utilities/python/Platform.java
package com.ultreon.commons.utilities.python;

import com.ultreon.commons.utilities.system.OS;
import com.ultreon.commons.utilities.system.SystemEnum;

public class Platform {
    private Platform() {

    }

    public static SystemEnum getSystem() {
        String sunDesktop = System.getProperty("os.name");
        if (sunDesktop.toLowerCase().startsWith("windows")) {
            return SystemEnum.WINDOWS;
        } else if (sunDesktop.equalsIgnoreCase("solaris")) {
            return SystemEnum.SOLARIS;
        } else if (sunDesktop.equalsIgnoreCase("android")) {
            return SystemEnum.ANDROID;
        } else if (sunDesktop.equalsIgnoreCase("ios")) {
            return SystemEnum.IOS;
        } else if (sunDesktop.equalsIgnoreCase("linux")) {
            return SystemEnum.LINUX;
        } else if (sunDesktop.equalsIgnoreCase("darwin")) {
            return SystemEnum.DARWIN;
        } else if (sunDesktop.equalsIgnoreCase("osx")) {
            return SystemEnum.OSX;
        } else if (sunDesktop.equalsIgnoreCase("mac os")) {
            return SystemEnum.OSX;
        } else if (sunDesktop.equalsIgnoreCase("mac os x")) {
            return SystemEnum.OSX;
        } else if (sunDesktop.equalsIgnoreCase("os/2")) {
            return SystemEnum.OS_2;
        } else if (sunDesktop.equalsIgnoreCase("aix")) {
            return SystemEnum.AIX;
        } else if (sunDesktop.equalsIgnoreCase("freebsd")) {
            return SystemEnum.FREE_BSD;
        } else if (sunDesktop.equalsIgnoreCase("irix")) {
            return SystemEnum.IRIX;
        } else if (sunDesktop.equalsIgnoreCase("digital unix")) {
            return SystemEnum.DIGITAL_UNIX;
        } else if (sunDesktop.equalsIgnoreCase("mpe/ix")) {
            return SystemEnum.MPE_IX;
        } else if (sunDesktop.equalsIgnoreCase("hp ux")) {
            return SystemEnum.HP_UX;
        } else if (sunDesktop.toLowerCase().startsWith("netware")) {
            return SystemEnum.NETWARE;
        } else if (sunDesktop.equalsIgnoreCase("dos")) {
            return SystemEnum.DOS;
        } else if (sunDesktop.equalsIgnoreCase("unix")) {
            return SystemEnum.UNIX;
        }
        return SystemEnum.UNKNOWN;
    }

    public static OS getOS() {
        return new OS();
    }
}

File: src/main/kotlin/qtech/utilities/python/Platform.kt
package qtech.utilities.python

import qtech.utilities.system.OS
import qtech.utilities.system.SystemEnum

object Platform {
    val system: SystemEnum
        get() {
            val sunDesktop = System.getProperty("os.name")
            when {
                sunDesktop.toLowerCase().startsWith("windows") -> {
                    return SystemEnum.WINDOWS
                }
                sunDesktop.equals("solaris", ignoreCase = true) -> {
                    return SystemEnum.SOLARIS
                }
                sunDesktop.equals("android", ignoreCase = true) -> {
                    return SystemEnum.ANDROID
                }
                sunDesktop.equals("ios", ignoreCase = true) -> {
                    return SystemEnum.IOS
                }
                sunDesktop.equals("linux", ignoreCase = true) -> {
                    return SystemEnum.LINUX
                }
                sunDesktop.equals("darwin", ignoreCase = true) -> {
                    return SystemEnum.DARWIN
                }
                sunDesktop.equals("osx", ignoreCase = true) -> {
                    return SystemEnum.OSX
                }
                sunDesktop.equals("mac os", ignoreCase = true) -> {
                    return SystemEnum.OSX
                }
                sunDesktop.equals("mac os x", ignoreCase = true) -> {
                    return SystemEnum.OSX
                }
                sunDesktop.equals("os/2", ignoreCase = true) -> {
                    return SystemEnum.OS_2
                }
                sunDesktop.equals("aix", ignoreCase = true) -> {
                    return SystemEnum.AIX
                }
                sunDesktop.equals("freebsd", ignoreCase = true) -> {
                    return SystemEnum.FREE_BSD
                }
                sunDesktop.equals("irix", ignoreCase = true) -> {
                    return SystemEnum.IRIX
                }
                sunDesktop.equals("digital unix", ignoreCase = true) -> {
                    return SystemEnum.DIGITAL_UNIX
                }
                sunDesktop.equals("mpe/ix", ignoreCase = true) -> {
                    return SystemEnum.MPE_IX
                }
                sunDesktop.equals("hp ux", ignoreCase = true) -> {
                    return SystemEnum.HP_UX
                }
                sunDesktop.toLowerCase().startsWith("netware") -> {
                    return SystemEnum.NETWARE
                }
                sunDesktop.equals("dos", ignoreCase = true) -> {
                    return SystemEnum.DOS
                }
                sunDesktop.equals("unix", ignoreCase = true) -> {
                    return SystemEnum.UNIX
                }
                else -> return SystemEnum.UNKNOWN
            }
        }
    val oS: OS
        get() = OS()
}
